python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -524,6 +531,72 @@ def resolve_hasResumePermission(self, graphene_info: ResolveInfo) -> bool:
     def resolve_user(self, _graphene_info: ResolveInfo) -> Optional[str]:
         return self._backfill_job.user
 
+    def resolve_logEvents(self, graphene_info: ResolveInfo, cursor: Optional[str] = None):
+        # very unsure how the cursor gets passed through, especially if called from get_backfills which also has a cursor
+        from ..schema.instigation import (
+            GrapheneInstigationEvent,
+            GrapheneInstigationEventConnection,
+        )
+        from ..schema.logs.log_level import GrapheneLogLevel
+
+        backfill_id = self._backfill_job.backfill_id
+        # TODO find a better way to keep this in sync. maybe store as part of the asset backfill data?
+        backfill_log_key_prefix = ["backfill", backfill_id]
+
+        instance = graphene_info.context.instance
+
+        if not isinstance(instance.compute_log_manager, CapturedLogManager):
+            return GrapheneInstigationEventConnection(events=[], cursor="", hasMore=False)
+
+        # TODO - need to gate to plus only
+
+        log_keys = sorted(
+            instance.compute_log_manager.get_log_keys_for_log_key_prefix(
+                backfill_log_key_prefix, io_type=ComputeIOType.STDERR
+            )
+        )
+        log_key_to_fetch = cursor.split("/") if cursor else log_keys[0]
+        log_data = instance.compute_log_manager.get_log_data(log_key_to_fetch)
+        raw_logs = log_data.stderr.decode("utf-8") if log_data.stderr else ""
+
+        records = []
+        for line in raw_logs.split("\n"):
+            if not line:
+                continue
+            try:
+                records.append(_seven.json.loads(line))
+            except json.JSONDecodeError:
+                continue
+
+        events = []
+        for record_dict in records:
+            exc_info = record_dict.get("exc_info")
+            message = record_dict.get("msg")
+            if exc_info:
+                message = f"{message}\n\n{exc_info}"
+            event = GrapheneInstigationEvent(
+                message=message,
+                level=GrapheneLogLevel.from_level(record_dict["levelno"]),
+                timestamp=int(record_dict["created"] * 1000),
+            )
+
+            events.append(event)
+        current_log_key_idx = log_keys.index(log_key_to_fetch)
+        has_more = current_log_key_idx < len(log_keys) - 1
+        if has_more:
+            next_log_key = (
+                log_keys[current_log_key_idx + 1]
+                if current_log_key_idx < len(log_keys) - 1
+                else None
+            )
+        else:
+            next_log_key = None  # TODO what should this be?
```

Comment:
when we hit the end of the logs, what should the cursor be? There are two cases to consider:
1. the backfill is in progress so even if we've hit the end of the logs, we may get more logs in the future
2. the backfill is over, the end of the logs is the true end of the logs